src/screens/Profile/PaymentOptions/index.tsx

```diff
@@ -0,0 +1,205 @@
+import {useAuthState} from '@atb/auth';
+import FullScreenHeader from '@atb/components/screen-header/full-header';
+import ThemeText from '@atb/components/text';
+import PaymentBrand from '@atb/screens/Ticketing/Purchase/Payment/PaymentBrand';
+import {
+  getExpireDate,
+  getPaymentTypeName,
+} from '@atb/screens/Ticketing/Purchase/utils';
+import {StyleSheet, Theme, useTheme} from '@atb/theme';
+import {
+  deleteRecurringPayment,
+  listRecurringPayments,
+  RecurringPayment,
+} from '@atb/tickets';
+import {useTranslation} from '@atb/translations';
+import PaymentOptionsTexts from '@atb/translations/screens/subscreens/PaymentOptions';
+import React, {useEffect, useState} from 'react';
+import {ActivityIndicator, TouchableOpacity, View} from 'react-native';
+import {ScrollView} from 'react-native-gesture-handler';
+import {destructiveAlert} from '../Home/utils';
+import MessageBox from '@atb/components/message-box';
+import {StackNavigationProp} from '@react-navigation/stack';
+import {RootStackParamList} from '@atb/navigation';
+import useFontScale from '@atb/utils/use-font-scale';
+
+type PaymentOptionsProps = {
+  navigation: StackNavigationProp<RootStackParamList>;
+};
+
+export default function PaymentOptions({navigation}: PaymentOptionsProps) {
+  const style = useStyle();
+  const {theme} = useTheme();
+  const {t} = useTranslation();
+  const {user} = useAuthState();
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [storedCards, setStoredCards] = useState<RecurringPayment[]>([]);
+
+  async function getRecurringPaymentOptions(): Promise<
+    Array<RecurringPayment>
+  > {
+    if (!user || user.isAnonymous) return [];
+    const recurringOptions: Array<RecurringPayment> =
+      await listRecurringPayments();
+    return [...recurringOptions.reverse()];
+  }
+
+  /*
+   * Refresh stored cards when screen gets focus
+   */
+  useEffect(() => {
+    const unsubscribe = navigation.addListener('focus', refreshCards);
+    return unsubscribe;
+  }, [navigation]);
+
+  async function handleRemovePayment(paymentOption: RecurringPayment) {
+    setIsLoading(true);
+    await deleteRecurringPayment(paymentOption.id);
+    refreshCards();
+  }
+
+  async function refreshCards() {
+    setIsLoading(true);
+    const remoteOptions = await getRecurringPaymentOptions();
+    setStoredCards(remoteOptions);
+    setIsLoading(false);
+  }
+
+  return (
+    <View style={style.container}>
+      <FullScreenHeader
+        title={t(PaymentOptionsTexts.header.title)}
+        leftButton={{type: 'back'}}
+      />
+
+      {!isLoading && storedCards.length == 0 && (
+        <View accessibilityLiveRegion="polite">
+          <MessageBox
+            message={t(PaymentOptionsTexts.noStoredCards)}
+            withMargin={true}
+            containerStyle={{marginTop: theme.spacings.xLarge}}
+          />
+        </View>
+      )}
+
+      <ScrollView style={style.contentMargin}>
```

Comment:
If we used the FlatList instead we'd get pull to refresh which I guess is a more common way to refresh data on apps

Reply:
I don't see the scenario where the user would want to refresh manually, or have any reason to do so.
The reason we do a refresh on focus is to sync if the user navigates *back* to "removePaymentOptions" after storing a new card, because the screen would steill be mounted.